vllm/core/block_manager.py

```diff
@@ -98,6 +98,41 @@ def get_num_free_blocks(self) -> int:
         return (self.num_blocks - self.current_num_blocks +
                 self.evictor.num_blocks)
 
+    def get_num_cached_blocks(self, seq: Sequence) -> int:
+        if not self.enable_caching:
+            return 0
+        num_cached_blocks = 0
+        for logical_idx in range(len(seq.logical_token_blocks)):
+            block_hash = seq.hash_of_block(logical_idx)
+            if block_hash in self.cached_blocks:
+                num_cached_blocks += 1
+        return num_cached_blocks
+
+    def get_num_computed_blocks(self, seq: Sequence) -> int:
+        if not self.enable_caching:
+            return 0
+        num_computed_blocks = 0
+        for logical_idx in range(len(seq.logical_token_blocks)):
+            block_hash = seq.hash_of_block(logical_idx)
+            if block_hash in self.cached_blocks:
+                if self.cached_blocks[block_hash].computed:
+                    num_computed_blocks += 1
+                else:
+                    break
```

Comment:
Comment why it breaks? (I assume it has assumption blocks are computed in order?)